R/runtime-caches.R

```diff
@@ -0,0 +1,73 @@
+#' Get runtime caches
+#' 
+#' View the runtime caches on a Connect server.
+#' 
+#' @param client A `Connect` object.
+#' 
+#' @returns A tibble of runtime caches on the server, showing `language`,
+#' `version` and `image_name`. For Connect servers not using off-host execution,
+#' `image_name` is `"Local"`.
+#' 
+#' @examples
+#' \dontrun{
+#' client <- connect()
+#' get_runtime_caches(client)
+#' }
+#' 
+#' @family server management functions
+#' @export
+get_runtime_caches <- function(client) {
+  validate_R6_class(client, "Connect")
+  res <- client$GET(v1_url("system/caches/runtime"))
+  purrr::map_dfr(res$caches, ~.x)
+}
+
+#' Delete a runtime cache
+#' 
+#' Delete a runtime cache from a Connect server.
+#' 
+#' @param client A `Connect` object.
+#' @param language The language of the cache, either "R" or "Python".
+#' @param version The version of the cache, e.g. "3.6.3".
+#' @param image_name Optional. The name of the off-host execution image for
+#'   the cache, or "Local" (the default) for native execution caches.
+#' @param dry_run Optional, default `FALSE`. If true, perform a dry run of
+#'   the deletion.
+#' 
+#'
+#' @returns A `Task` object representing the deletion task. If `dry_run` is
+#' `TRUE`, returns `NULL` or throws an error if the deletion would fail.
+#'
+#' @examples
+#' \dontrun{
+#' client <- connect()
+#' task <- delete_runtime_cache(client, "R", "3.6.3")
+#' poll_task(task)
+#' }
+#' 
+#' @family server management functions
+#' @export
+delete_runtime_cache <- function(client, language, version, image_name = "Local", dry_run = FALSE) {
+  res <- client$DELETE(
+    path = v1_url("system/caches/runtime"),
+    body = list(
+      language = language,
+      version = version,
+      image_name = image_name,
+      dry_run = dry_run
+    ),
+    encode = "json",
+    parser = "parsed"
+  )
+
+  if (dry_run == TRUE) {
+    message(glue::glue(
+      "Runtime cache deletion dry run finished; ",
+      "language = {language}, version = {version}, image_name = {image_name}"
+    ))
```

Comment:
I don't know how other similar parts of the code work...

Is there anything from `res` that can be included here?

Reply:
There isn't really anything meaningful in the response — it's basically a `200` if the directory exists and has the requisite permissions to delete (I think), and an error if not (and we report the error). The success is pretty information-free. Happy to take suggestions for better wording though. This wording is taken from the `rsconnect-python` implementation.

Reply:
I think I'll change it to say "did not encounter any errors", as that's kind of a better representation of what it means.